src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -4072,7 +4079,7 @@ void MacroAssembler::verify_secondary_supers_table(Register r_sub_klass,
                                                    Register tmp1,
                                                    Register tmp2,
                                                    Register tmp3) {
-  assert_different_registers(r_sub_klass, r_super_klass, tmp1, tmp2, tmp3, result, t0);
+  assert_different_registers(r_sub_klass, r_super_klass, tmp1, tmp2, tmp3, result, t0, t1);
```

Comment:
This one is used by `MacroAssembler::lookup_secondary_supers_table` which is called by `instruct partialSubtypeCheckConstSuper` in riscv.ad. We should list the rflags as being killed for this instruct. As this is not obvious, we should double check if there are other similar occurrences like this.

Reply:
Same anwser here actual call do not need to kill CR as RFLAG is SOC.

Reply:
Note that C2 instruct `partialSubtypeCheckConstSuper` is not a call. So this issue is still there for this case I think.

Reply:
Note sure what you mean by "t0 and t1 is reserved from C2".
They are not allocatable by register allocator.
But C2 think there is RFLAG register (still not allocatable), which we have aliased with x6/t1.
This pseudo register will be SOC.
Any live SOC should be pushed before a call.

The clobbering we added for CallRuntime/CallDynamicJava is not needed AFAIK, check e.g. x86 it do not kill CR as RFLAG is also set to SOC on x86 (but here backed with a real register).

Reply:
> Note sure what you mean by "t0 and t1 is reserved from C2". They are not allocatable by register allocator. But C2 think there is RFLAG register (still not allocatable), which we have aliased with x6/t1. This pseudo register will be SOC. Any live SOC should be pushed before a call.

Yeah, they are not allocatable. That's what I mean by saying they are reserved :-)

> The clobbering we added for CallRuntime/CallDynamicJava is not needed AFAIK, check e.g. x86 it do not kill CR as RFLAG is also set to SOC on x86 (but here backed with a real register).

Yes, I also checked other CPU platforms and I think you are right in that these effect of killing CR is unnecessary. Maybe a separate PR to clean them up.

Reply:
New PR for this cleanup: https://github.com/openjdk/jdk/pull/21576